var/spack/repos/builtin/packages/mapl/package.py

```diff
@@ -119,3 +119,12 @@ def patch(self):
               subprocess.check_output(nc_pc_cmd, encoding="utf8").strip()
             filter_file("(target_link_libraries[^)]+PUBLIC )", \
               r'\1 %s '%nc_flags, "pfio/CMakeLists.txt")
+
+    def setup_build_environment(self, env):
+        # This is really, really bad. esma_cmake, an internal
+        # dependency of mapl, is looking for the cmake argument
+        # -DBASEDIR, and if it doesn't find it, it's looking
+        # for an environment variable with the same name. This
+        # name is common and used all over the place, and if
+        # it is set it breaks the mapl build. Unset it here ...
+        env.unset("BASEDIR")
```

Comment:
Oops. Sorry about this, @climbfuji. I added that code because users here didn't want to always pass in `-DBASEDIR` on the CMake line. I don't mind doing it, but users liked being able to do `cmake -B build` and all works.

Reply:
Not a problem at all, just wondering if we could use a safer name going forward, for example GEOS_BASEDIR or something like that?